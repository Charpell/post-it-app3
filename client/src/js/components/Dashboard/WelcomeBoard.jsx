import React, { Component } from 'react';
import toastr from 'toastr';
import moment from 'moment';

import Message from './Message';
import AppActions from '../../actions/AppActions';
import AppStore from '../../stores/AppStore';

/**
 * @description The Presentation component that servers all message activities
 * 
 * @class MessageBoard
 * 
 * @extends {Component}
 */
export default class WelcomeBoard extends Component {

    render() {
        let userName = JSON.parse(localStorage.getItem('user'));
        return (
            <div className="container" id="main">
                <div className="row">
                    <div className="col-md-12" data-intro='Message Board where every messages are posted and received. Happy Posting!' >
                        <div className="viewMessageBoard">
                            <div className="container" >
                                <div className="row">
                                    <div className="col-sm-3"></div>
                                    <div className="col-md-6 col-sm-6 col-xs-12">                                       
                                        <center>
                                        <h3>Welcome {userName}</h3>
                                        </center>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

        )
    }

}
