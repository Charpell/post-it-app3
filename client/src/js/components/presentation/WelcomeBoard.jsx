import React from 'react';


/**
 * @description The Default Page when the user logs in
 *
 * @class WelcomeBoard
 *
 * @extends {Component}
 */
const WelcomeBoard = () => ({
  render() {
    return (
        <div className="container" id="main">
            <div className="row">
                <div className="col-md-12" >
                    <div className="viewMessageBoard">
                        <div className="container" >
                        <div className="row">
                            <div className="col-sm-3"></div>
                            <div className="col-md-6 col-sm-6 col-xs-12">
                                <center>
                                    <h3>Welcome {this.props.userName}</h3>
                                </center>
                            </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
  }
});

export default WelcomeBoard;
