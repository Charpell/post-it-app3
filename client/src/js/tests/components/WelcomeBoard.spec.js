import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import WelcomeBoard from '../../components/Dashboard/WelcomeBoard'


describe('WelcomeBoard component', () => {
  it('WelcomeBoard component should render as expected', () => {
    const tree = renderer.create(<WelcomeBoard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});