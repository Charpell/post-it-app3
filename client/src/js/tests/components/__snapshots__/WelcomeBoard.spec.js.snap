// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeBoard component WelcomeBoard component should render as expected 1`] = `
<div
  className="container"
  id="main"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <div
        className="viewMessageBoard"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-3"
            />
            <div
              className="col-md-6 col-sm-6 col-xs-12"
            >
              <center>
                <h3>
                  Welcome 
                </h3>
              </center>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
